pkg/util/distance/calc_distance.go

```diff
@@ -130,15 +131,23 @@ func CalcFloatDistance(dim int64, left, right []float32, metric string) ([]float
 
 	distArray := make([]float32, leftNum*rightNum)
 
-	// Multi-threads to calculate distance. TODO: avoid too many go routines
+	// Multi-threads to calculate distance.
 	var waitGroup sync.WaitGroup
-	CalcWorker := func(index int64) {
+	poolSize := int(leftNum / 3)
+	pool, err := ants.NewPoolWithFunc(poolSize, func(i interface{}) {
```

Comment:
pool.Release() is not been called?

Reply:
Would it be acceptable to continue using the previous approach, but submit a group of coroutines in batches?

Reply:
ok, i am accept and thanks for your review.

Reply:
@CocaineCong could you reopen or submit a new pr by the suggestion above?

Reply:
ok，i will resubmit this pr later！